site/src/component/CoursePopover/CoursePopover.tsx

```diff
@@ -26,6 +28,11 @@ const CoursePopover: FC<CoursePopoverProps> = ({ course, interactive = true, req
     </div>
   );
 
+  const clearedCourses = useClearedCourses();
+  if (typeof course !== 'string') {
+    requiredCourses = getMissingPrerequisites(clearedCourses, course);
+  }
```

Comment:
same if condition as the next one; combine the two.

Additionally, if we're always going override the `requiredCourses` array when the data is done loading, then I think we should just change `requiredCourses` to be a local variable instead of a prop passed in.